x/app-router/module_ibc.go

```diff
@@ -0,0 +1,364 @@
+package app_router
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v3/modules/core/05-port/types"
+
+	"github.com/Stride-Labs/stride/v3/x/app-router/keeper"
+	"github.com/Stride-Labs/stride/v3/x/app-router/types"
+
+	// "google.golang.org/protobuf/proto" <-- this breaks tx parsing
+
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+)
+
+// IBC MODULE IMPLEMENTATION
+// IBCModule implements the ICS26 interface for transfer given the transfer keeper.
+type IBCModule struct {
+	keeper keeper.Keeper
+	app    porttypes.IBCModule
+}
+
+// NewIBCModule creates a new IBCModule given the keeper
+func NewIBCModule(k keeper.Keeper, app porttypes.IBCModule) IBCModule {
+	return IBCModule{
+		keeper: k,
+		app:    app,
+	}
+}
+
+// OnChanOpenInit implements the IBCModule interface
+func (im IBCModule) OnChanOpenInit(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID string,
+	channelID string,
+	channelCap *capabilitytypes.Capability,
+	counterparty channeltypes.Counterparty,
+	version string,
+) error {
+	// Note: The channel capability must be claimed by the authentication module in OnChanOpenInit otherwise the
+	// authentication module will not be able to send packets on the channel created for the associated interchain account.
+	// NOTE: unsure if we have to claim this here! CHECK ME
```

Comment:
Is this still an open question? I see it in `x/records` too

Reply:
Hmm, will need to wait for Aidan's message

Reply:
@asalzmann

Reply:
no need to claim here as the transfer module owns the channel